src/xsdba/nbutils.py

```diff
@@ -181,17 +181,19 @@ def vecquantiles(da: DataArray, rnk: DataArray, dim: str | Sequence[Hashable]) -
     xarray.DataArray
         The quantiles computed along the `dim` dimension.
     """
-    tem = utils.get_temp_dimname(da.dims, "temporal")
     dims = [dim] if isinstance(dim, str) else dim
+    common_dims = list(set(da.dims).intersection(set(rnk.dims)))
+    tem = utils.get_temp_dimname(da.dims, "temporal")
     da = da.stack({tem: dims})
-    da = da.transpose(*rnk.dims, tem)
-
-    res = DataArray(
-        _vecquantiles(da.values, rnk.values),
-        dims=rnk.dims,
-        coords=rnk.coords,
-        attrs=da.attrs,
-    ).astype(da.dtype)
+
+    res = apply_ufunc(
+        _vecquantiles,
+        da,
+        rnk,
+        input_core_dims=[common_dims + [tem], common_dims],
```

Comment:
this leaves out the possibility that `da` or `rnk` could have more dimensions than the other one. E.g. `da` has `add_dims` dimensions. So we will compute the quantiles for each different `add_dims`.